File: src/main/java/compiler/parser/ifs/IfParser.java
package compiler.parser.ifs;

import compiler.block.Block;
import compiler.block.ifs.IfBlock;
import compiler.parser.Parser;
import compiler.tokenizer.Token;
import compiler.tokenizer.TokenData;
import compiler.tokenizer.TokenType;
import compiler.tokenizer.Tokenizer;

/*
need to make parameter single variable names instead as cant define a variable in an ifs...
 */


public class IfParser extends Parser<IfBlock> {
    @Override
    public boolean shouldParse(String line) {
        return line.matches("if \\((.*)*\\):");
    }

    @Override
    public IfBlock parse(Block superBlock, Tokenizer tokenizer) {

        tokenizer.nextToken();  //skip if
        tokenizer.nextToken();  // skip (
        Token first = tokenizer.nextToken();
        Token operator = tokenizer.nextToken();

        if (operator.getToken().equals(")")) {
            return new IfBlock(superBlock, first.getToken());
        } else {
            Token value = tokenizer.nextToken();
            return new IfBlock(superBlock, first.getToken() + " " + operator.getToken() + " " + value.getToken());
        }
    }
}

File: src/main/scala/compiler/parser/ifs/IfParser.scala
package compiler.parser.ifs

import compiler.block.Block
import compiler.block.ifs.IfBlock
import compiler.block.loops.WhileBlock
import compiler.parser.Parser
import compiler.tokenizer.Tokenizer
import test_classes.block.Block
import test_classes.block.ifs.IfBlock
import test_classes.parser.Parser
import test_classes.tokenizer.Tokenizer

class IfParser extends Parser[IfBlock] {

  def shouldParse(line: String): Boolean = line.matches("if \\((.*)*\\):")

  def parse(superBlock: Block, tokenizer: Tokenizer): IfBlock = {
    tokenizer.nextToken //skip if
    tokenizer.nextToken // skip (

    // Loop through getting each parameter and  adding to a String
    var statement: String = ""
    var nextToken: String = tokenizer.nextToken.token
    while (nextToken != ")") {
      {
        if (nextToken == "=") statement += nextToken
        else statement += " " + nextToken + " "
        nextToken = tokenizer.nextToken.token
      }
    }
    return new IfBlock(superBlock, statement.trim.replaceAll(" +", " "))
  }
}
